### ISSUE TITLE
Error installing google_auth extra on 1.10.13

### ISSUE THREAD

**Issue by jonathonbattista** [apache/airflow#12759]
**Apache Airflow version**: 1.10.13

**Environment**: Docker

**What happened**:
Looks like there are conflicts between the setup.py (which limits to oauthlib!=2.0.3,!=2.0.4,!=2.0.5,<3.0.0,>=1.1.2 and requests-lib==1.1.0 ) and the version specified in the requirements-3.7.txt (which is oauthlib==3.1.0 and request-oauthlib===1.3.0 )

- [https://github.com/apache/airflow/blob/1.10.13/setup.py](https://github.com/apache/airflow/blob/1.10.13/setup.py)
- [https://raw.githubusercontent.com/apache/airflow/constraints-1.10.13/constraints-3.7.txt](https://raw.githubusercontent.com/apache/airflow/constraints-1.10.13/constraints-3.7.txt)

oauthlib: 
- ERROR: Could not find a version that satisfies the requirement oauthlib!=2.0.3,!=2.0.4,!=2.0.5,<3.0.0,>=1.1.2; extra == "google_auth" (from apache-airflow[google-auth])
- ERROR: No matching distribution found for oauthlib!=2.0.3,!=2.0.4,!=2.0.5,<3.0.0,>=1.1.2; extra == "google_auth"

requests-oauthlib:
- ERROR: Could not find a version that satisfies the requirement requests-oauthlib==1.1.0; extra == "google_auth" (from apache-airflow[crypto,gcp,google-auth,kubernetes,mysql,s3,slack,statsd])
- ERROR: No matching distribution found for requests-oauthlib==1.1.0; extra == "google_auth"

**What you expected to happen**:

google_auth is installed successfully with constraints-3.7

**How to reproduce it**:
pip3 install --user 'apache-airflow[google_auth]==1.10.13' --constraint https://raw.githubusercontent.com/apache/airflow/constraints-1.10.13/constraints-3.7.txt

**Comment by potiuk** [comment]
Yep. Just before your issue I created those two:

https://github.com/oauthlib/oauthlib/issues/744
https://github.com/pypa/pip/issues/9203

and Updated documentation for upcoming 1.10.14. You need to downgrade PyPI to 20.2.4 or use `--use-deprecated legacy-resolver` switch when running pip install

**Comment by jonathonbattista** [comment]
@potiuk The error still persists in 1.10.14.
```
ERROR: Could not find a version that satisfies the requirement requests-oauthlib==1.1.0; extra == "google_auth" (from apache-airflow[,google-auth])
ERROR: No matching distribution found for requests-oauthlib==1.1.0; extra == "google_auth"
```
The **requests-oauthlib** version is still pinned to **1.1.0**  in the [setup.py](https://github.com/apache/airflow/blob/1.10.14/setup.py#L265)

**Comment by potiuk** [comment]
PLease use 1.10.14. Dependency problem has been solved there actually (including the oauthlib ones) and 1.10.13 is "bad" release - it has known problems with not started "depends_on_past" tasks. 1.10.14 has been officially released today.

**Comment by jonathonbattista** [comment]
You say the dependency problem has been solved in 1.10.14 but I that is not what I am seeing.

OAuthlib:
```
pip3 install --user 'apache-airflow[google_auth]==1.10.14' --constraint requirements-python3.7.txt
Collecting apache-airflow[google_auth]==1.10.14
 Downloading apache_airflow-1.10.14-py2.py3-none-any.whl (4.7 MB)
ERROR: Could not find a version that satisfies the requirement oauthlib!=2.0.3,!=2.0.4,!=2.0.5,<3.0.0,>=1.1.2; extra == "google_auth" (from apache-airflow[crypto,gcp,google-auth,kubernetes,mysql,s3,slack,statsd])
ERROR: No matching distribution found for oauthlib!=2.0.3,!=2.0.4,!=2.0.5,<3.0.0,>=1.1.2; extra == "google_auth"
```
requests-oauthlib:
```
ERROR: Could not find a version that satisfies the requirement requests-oauthlib==1.1.0; extra == "google_auth" (from apache-airflow[,google-auth])
ERROR: No matching distribution found for requests-oauthlib==1.1.0; extra == "google_auth"
```

**Comment by potiuk** [comment]
You did not try with the 1.10.14 constraints apparently

**Comment by potiuk** [comment]
Which PI0 version you use ? There are both gs in the latest released PIp 20.3 and PiP team is working on solving them . See https://github.com/apache/airflow/issues/12838 for details and workarounds
